lxd/storage/drivers/volume.go

```diff
@@ -115,7 +115,10 @@ func (v Volume) MountTask(task func(mountPath string, op *operations.Operation)
 	if isSnap {
 		unlock := lock(mountLockID)
 
-		ourMount, err := v.driver.MountVolumeSnapshot(v.volType, parentName, snapName, op)
+		// Override volume name
+		v.name = parentName
```

Comment:
@monstermunchkin this caught my eye, it looks like it could be dangerous as you're modifying the volume name that was defined by the user who created this Volume. The Volume could be used for other things after this `MountTask` call and would lead to unexpected results. @stgraber originally encourage the separation here of parent name and snap name rather than pass it as volume name (i.e "parentName/snapName"). This is another example why I didn't use `Volume` type for all function definitions. 

However if we go down this route for consistency (and overall I think it is a worthwhile endeavour) then really we should just update the `MountVolumeSnapshot` to accept a `Volume` type that is itself a snapshot and update the implementations to extract the parent name and snapshot name from the volume name.

Reply:
@monstermunchkin we would also need to mirror that change to `UnmountVolumeSnapshot` such that it just accepts a snapshot `Volume` and not a snapName arg.